astropy/coordinates/builtin_frames/utils.py

```diff
@@ -107,3 +107,32 @@ def get_dut1utc(time):
         msg = e.args[0] + ' Assuming UT1-UTC=0 for coordinate transformations.' + _IERS_HINT
         warnings.warn(msg, AstropyWarning)
         return np.zeros(time.shape)
+
+def get_jd12(time, scale):
+    """
+    Gets ``jd1`` and ``jd2`` from a time object in a particular scale.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        The time to get the jds for
+    scale : str
+        The time scale to get the jds for
+
+    Returns
+    -------
+    jd1 : float
+    jd2 : float
+    """
+    if time.scale == scale:
+        newtime = time
+    elif time.scale == 'ut1' or scale == 'ut1':
+        olddt = time.delta_ut1_utc
```

Comment:
By just accessing `delta_ut1_utc` you'll get an `IndexError` if the time is outside of the `IERS` table. I think if the time is OK, it is not a problem for the time instance to change, but if it is outside `IERS`, it should ideally not be set. I think you would need to do something like:
```
    if scale == time.scale:
        newtime = time
    else:
        try:
            newtime = getattr(time, scale)
        except IndexError:
            # similar warning to that in get_dut1dutc
            newtime = time
    return newtime.jd1, newtime.jd2
```